lib/validators/calculating_smoking_gun_validator.rb

```diff
@@ -0,0 +1,82 @@
+module Validators
+  class CalculatingSmokingGunValidator < SmokingGunValidator
+    include Validators::Validator
+    def initialize(measures, records, test_id)
+      if measures.length > 1
+        # throw some kind of an error here because we only want to do this for
```

Comment:
but the error would be because the software did something it wasn't
supposed to do not the user , so I dont think add_error is appropriate in
this case.

On Tue, Jan 19, 2016 at 2:25 PM, okeefm <notifications@github.com> wrote:

> In lib/validators/calculating_smoking_gun_validator.rb
> <https://github.com/projectcypress/cypress/pull/298#discussion_r50161076>:
>
> > @@ -0,0 +1,82 @@
> > +module Validators
> > +  class CalculatingSmokingGunValidator < SmokingGunValidator
> > +    include Validators::Validator
> > +    def initialize(measures, records, test_id)
> > +      if measures.length > 1
> > +        # throw some kind of an error here because we only want to do this for
>
> I think add_error should work in this context, so could we throw an error
> like "Too many measures provided for a single-measure test" in the
> initializer? Not sure if we would then want to not validate the first
> measure, or continue to evaluate it anyways.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/projectcypress/cypress/pull/298/files#r50161076>.
>